src/gordon_gcp/plugins/service/enricher.py

```diff
@@ -195,6 +197,27 @@ def _get_internal_ip(self, instance_data):
                 fqdn, external_ip, default_ttl, msg_logger)
         ]
 
+    async def _get_matching_records(self, instance_resource_url):
+        # Get the fqdn of the instance
+        instance_name = instance_resource_url.split('/')[-1]
+        fqdn = self._get_fqdn(instance_name)
+
+        # Get all Google DNS records for the managed zone
+        project = self.config.get('project')
+        managed_zone = self.config['managed_zone']
+        resource_records_url = GCEEnricher.RESOURCE_RECORDS_ENDPOINT.format(
+            project=project, managedZone=managed_zone)
+        search_params = {'name': fqdn}
+        response = await self._http_client.get_all(
+            resource_records_url, params=search_params)
+
+        # Get the Google DNS records with name matching the instance fqdn
+        matching_records = []
+        for resp in response:
```

Comment:
This may not need to be a loop - there should only be at most one response (I think?)

Reply:
I don't think there's any harm in keeping this as a loop, since there's no guarantee that the user has not added a bunch of records for that fqdn and that the response from `get_all` could include multiple pages.

Though on a similar note, do we want to ensure that the behavior for Gordon on deletion of a GCE instance **only** deletes records that Gordon itself created (in this case it would be just the A record), or will both delete the A record that Gordon itself created but also any records in Google Cloud DNS with the instance's fqdn as the record name?

example:
- `test-host-fdsa2.example.com` GCE instance is created
-  Gordon creates the A record for `test-host-fdsa2.example.com`
- someone puts into Cloud DNS a bunch of `TXT` records for `test-host-fdsa2.example.com`
- `test-host-fdsa2.example.com` GCE instance is deleted
- Which of these should the behavior for Gordon be:
   - Delete just the A record for `test-host-fdsa2.example.com`
   - Delete all the records for `test.example.com` (A record + all TXT records)

Reply:
I would expect Gordon to only delete records that it creates.

Reply:
@RochesterinNYC if "someone puts into Cloud DNS a bunch of `TXT` records" is done by manual record addition (i.e. via GHE), then I wouldn't expect gordon to delete them without first the user also making a pull request to GHE to remove them.

Reply:
Thanks for the input y'all. Switched to using `get_json` instead of `get_all` and now filter for both `name` and `type` of just `A` (since all we're creating are A records at the moment). Added a comment to document the fact that the types of records we return in `_get_matching_records_for_deletion` should match the types created in `_create_rrecords`.